### PR TITLE
Register mandatory and non-mandatory env vars

### LINKED ISSUE #21: Environment variable mandatory
There are some environment variable mandatory for starting web project. Most of them, shouldn't mandatory tough. I'd like to propose addition a fallback for each one env that is not really mandatory to start the application, such as `PUBLIC_HTML=/public`, `MONGODB_URI=mongodb://127.0.0.1:27017/db` and `PORT=8080`

@danielfireman What do you think ? Is it makes sense ?

### PR DESCRIPTION
**Pull request by lucasgomide**
This commit closes [issue #21](https://github.com/danielfireman/temp-to-go/issues/21)

---
This change is [Reviewable](https://reviewable.io/reviews/danielfireman/temp-to-go/22)

### CODE REVIEW

File: server/web/main.go [hunk 1]
```diff
@@ -21,29 +21,37 @@ const (
 	restrictedPath = "/restricted"
 )
 
+// Specification represents a map of enviornment variables
+type Specification struct {
+	UserPassword  string `envconfig:"USERPASSWD" required:"true"`
+	EncryptionKey string `split_words:"true" required:"true"`
```
Comment (danielfireman) [inline server/web/main.go]: Nit: Better to use `envconfig:"ENCRYPTION_KEY"` than split_words (which ends up doing the same thing, but indirectly).

Same bellow.

File: server/web/main.go [hunk 2]
```diff
@@ -21,29 +21,37 @@ const (
...
-	userPasswd := os.Getenv("USERPASSWD")
-	if userPasswd == "" {
-		log.Fatal("USERPASSWD must be set.")
+	var spec Specification
+	err := envconfig.Process("", &spec)
+	if err != nil {
+		log.Fatal(err.Error())
 	}
-	key := []byte(os.Getenv("ENCRYPTION_KEY"))
+
```
Comment (danielfireman) [inline server/web/main.go]: Log the environment variables/config?

fmt.Printf("%+v", spec) ?

**Review by danielfireman — CHANGES_REQUESTED** [review]
Super cool stuff!

Only small changes below. Furthermore, I believe you need to run `dep ensure` to update Gopkg files as you introduced `envconfig`.

File: server/web/main.go (hunk 2 above)
Comment (lucasgomide) [reply server/web/main.go]: I forgot to log it.. dammit

**Comment by danielfireman** [comment]
Thanks a lot, @lucasgomide ! :D

**Pull request merged**